python/paddle/fluid/layers/detection.py

```diff
@@ -3716,26 +3741,52 @@ def distribute_fpn_proposals(fpn_rois,
                 refer_level=4,
                 refer_scale=224)
     """
+    num_lvl = max_level - min_level + 1
+
+    if in_dygraph_mode():
+        assert rois_num is not None, "rois_num should not be None in dygraph mode."
```

Comment:
I suggest adding the reason why rois_num should not be None in error message.
Otherwise, users may be confused when seeing the error.

Not urgent, you can refine it in the next PR. Same for other APIs.